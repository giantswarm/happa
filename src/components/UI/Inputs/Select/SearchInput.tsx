import { Box } from 'grommet';
import * as React from 'react';

import TextInput from '../TextInput';

interface ISearchInputProps
  extends React.ComponentPropsWithoutRef<typeof TextInput> {}

const SearchInput: React.FC<ISearchInputProps> = (props) => {
  return (
    <Box pad={{ horizontal: 'xsmall', vertical: 'small' }} background='border'>
      <TextInput
        {...props}
        icon={<i className='fa fa-search' />}
        margin='none'
        size='small'
      />
    </Box>
  );
};

export default SearchInput;
