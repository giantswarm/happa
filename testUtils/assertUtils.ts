import { MatcherFunction, Nullish } from '@testing-library/react';

type Query = (matcher: MatcherFunction) => HTMLElement;

/**
 * Decorate an existing query to match text across multiple
 * elements.
 * @param query
 */
export function withMarkup(query: Query) {
  return (text: string) => {
    return query((_, node: Nullish<Element>) => {
      if (!node) return false;

      const hasText = (n: Element) => n.textContent === text;
      const hasTextInChildren = Array.from(node.children).some((child) =>
        hasText(child)
      );

      return hasText(node) && !hasTextInChildren;
    });
  };
}
