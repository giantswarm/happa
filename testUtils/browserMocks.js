// eslint-disable-next-line @typescript-eslint/no-var-requires
const fetch = require('isomorphic-fetch');

const localStorageMock = (function () {
  let store = {};

  return {
    getItem: function (key) {
      return store[key] || null;
    },
    setItem: function (key, value) {
      store[key] = value;
    },
    removeItem: function (key) {
      delete store[key];
    },
    clear: function () {
      store = {};
    },

    // replaceWith is not part of the localStorage spec, but useful for us
    // in testing to set the full localStorage to a known set of values.
    replaceWith: function (newStore) {
      store = newStore;
    },
  };
})();

Object.defineProperty(window, 'localStorage', {
  value: localStorageMock,
});

// Adjust the console logging behaviour during test runs.
/* eslint-disable no-console */
global.console = {
  log: console.log,
  error: console.error,
  warn: jest.fn(), // warnings are surpressed.
  info: console.info,
  debug: console.debug,
};
/* eslint-enable no-console */

global.fetch = fetch;
