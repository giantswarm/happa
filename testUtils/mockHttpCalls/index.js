/***** Constants *****/
export * from './constantsAndHelpers';

/***** Responses *****/

// Info
export * from './info';

// User
export * from './user';

// Auth
export * from './auth';

// Clusters
export * from './clusters';

// Organizations
export * from './organizations';

// Status
export * from './status';

// Apps
export * from './apps';

// Node Pools
export * from './nodePools';

// Releases
export * from './releases';

// App Catalogs
export * from './appCatalogs';
