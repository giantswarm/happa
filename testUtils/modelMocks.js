import * as authSelectors from 'selectors/authSelectors';

jest.mock('model/services/giantSwarm');
jest.mock('selectors/authSelectors');

// eslint-disable-next-line no-import-assign
authSelectors.selectAuthToken = jest.fn(
  (_, state) => state.app.loggedInUser.token
);
