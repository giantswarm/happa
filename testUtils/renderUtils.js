import { render } from '@testing-library/react';
import App from 'App';
import { ConnectedRouter } from 'connected-react-router';
import { ThemeProvider } from 'emotion-theming';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { AppRoutes } from 'shared/constants/routes';
import configureStore from 'stores/configureStore';
import theme from 'styles/theme';

export const initialStorage = {
  user:
    '{"email":"developer@giantswarm.io","auth":{"scheme":"giantswarm","token":"a-valid-token"},"isAdmin":true}',
};

/**
 * This function will render the whole app with a mocked store in the route
 * provided.
 * @param {String} route.
 * @param {HTMLElement} container An HTMLElement.
 */
export function renderRouteWithStore(
  initialRoute = AppRoutes.Home,
  state = {},
  storage = initialStorage
) {
  localStorage.replaceWith(storage);

  const history = createMemoryHistory({
    initialEntries: [initialRoute],
    initialIndex: 0,
  });

  const store = configureStore(state, history);

  const app = render(<App {...{ store, theme, history }} />);

  return app;
}

/**
 * Render component in the Theme context
 * @param {React.ReactType} component The React Component
 * @param {<P extends Record<string, any>>React.ComponentProps<P>} props Props to pass to the component
 * @param {<Q extends TestingLibrary.Queries>TestingLibrary.RenderOptions<Q>} [options] Testing library render options
 */
export function renderWithTheme(component, props, options) {
  return render(getComponentWithTheme(component, props), options);
}

/**
 * Include component in the Theme context and return it
 * @param {React.ReactType} Component The React Component
 * @param {<P extends Record<string, any>>React.ComponentProps<P>} props Props to pass to the component
 */
export function getComponentWithTheme(Component, props) {
  return (
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>
  );
}

/**
 * Render component in the Store context
 * @param {React.ReactType} component The React Component
 * @param {<P extends Record<string, any>>React.ComponentProps<P>} props Props to pass to the component
 * @param {Record<string, any>} state Current Store state
 * @param {<Q extends TestingLibrary.Queries>TestingLibrary.RenderOptions<Q>} [options] Testing library render options
 */
export function renderWithStore(component, props, state, options) {
  return render(getComponentWithStore(component, props, state), options);
}

/**
 * Include component in the Store context and return it
 * @param {React.ReactType} Component The React Component
 * @param {<P extends Record<string, any>>React.ComponentProps<P>} props Props to pass to the component
 * @param {Record<string, any>} state Current Store state
 * @param {Record<string, any>} storage Current LocalStorage value
 * @param {History} history Current Browser history
 */
export function getComponentWithStore(
  Component,
  props = {},
  state = {},
  storage = initialStorage,
  history = createMemoryHistory()
) {
  localStorage.replaceWith(storage);

  const store = configureStore(state, history);

  const app = (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <ConnectedRouter history={history}>
          <Component {...props} />
        </ConnectedRouter>
      </ThemeProvider>
    </Provider>
  );

  return app;
}
