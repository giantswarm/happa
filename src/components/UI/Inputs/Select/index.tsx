import {
  FormField,
  FormFieldProps,
  Select as Input,
  ThemeContext,
  ThemeType,
} from 'grommet';
import PropTypes from 'prop-types';
import * as React from 'react';
import { css } from 'styled-components';

import SearchInput from './SearchInput';

const customTheme: ThemeType = {
  global: {
    drop: {
      background: 'none',
      shadowSize: 'none',
      border: {
        radius: '0',
      },
      zIndex: '9999',
    },
  },
  select: {
    background: 'none',
    icons: {
      color: { dark: 'text', light: 'text' },
    },
    // @ts-expect-error
    searchInput: SearchInput,
    container: {
      // @ts-expect-error
      extend: css`
        background: ${({ theme }) => theme.global.colors['input-background']};
        border-bottom-left-radius: 4px;
        border-bottom-right-radius: 4px;
        border: 1px solid ${({ theme }) => theme.global.colors.border.dark};
      `,
    },
    control: {
      // @ts-expect-error
      extend: css`
        background: ${({ theme }) => theme.global.colors['input-background']};
        border-radius: 4px;
      `,
      open: css`
        border-bottom-left-radius: 0;
        border-bottom-right-radius: 0;
      `,
    },
    step: 5,
  },
  textInput: {
    // @ts-expect-error
    extend: css`
      font-size: ${({ theme }) => theme.global.font.size};
      line-height: ${({ theme }) => theme.global.font.height};
    `,
  },
};

interface ISelectProps extends React.ComponentPropsWithoutRef<typeof Input> {
  /**
   * Props to be passed to the text input wrapper.
   */
  contentProps?: FormFieldProps['contentProps'];
  /**
   * Props to be passed to the form field.
   */
  formFieldProps?: FormFieldProps & { className?: string };
  /**
   * Whether the input is required or not.
   */
  required?: FormFieldProps['required'];
  /**
   * An error to be displayed below the input.
   */
  error?: FormFieldProps['error'];
  /**
   * A description to be displayed below the input.
   */
  info?: FormFieldProps['info'];
  /**
   * A help text to be displayed between the label
   * and the input.
   */
  help?: FormFieldProps['help'];
  /**
   * The margin between the form field and other elements.
   */
  margin?: FormFieldProps['margin'];
  /**
   * Whether the input should have some extra padding or not.
   */
  pad?: FormFieldProps['pad'];
  /**
   * The description text displayed above the input.
   */
  label?: FormFieldProps['label'];
}

const Select = React.forwardRef<HTMLSelectElement, ISelectProps>(
  (
    {
      id,
      label,
      contentProps,
      formFieldProps,
      disabled,
      required,
      error,
      info,
      help,
      name,
      margin,
      pad,
      ...props
    },
    ref
  ) => {
    return (
      <ThemeContext.Extend value={customTheme}>
        <FormField
          htmlFor={id}
          contentProps={contentProps}
          disabled={disabled === true}
          required={required}
          name={name}
          error={error}
          info={info}
          help={help}
          margin={margin}
          pad={pad}
          label={label}
          {...formFieldProps}
        >
          <Input
            {...props}
            id={id}
            ref={ref as React.Ref<HTMLSelectElement> & string}
            disabled={disabled}
            name={name}
          />
        </FormField>
      </ThemeContext.Extend>
    );
  }
);

Select.propTypes = {
  name: PropTypes.string,
  id: PropTypes.string,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  contentProps: PropTypes.object,
  formFieldProps: PropTypes.object,
  disabled: PropTypes.bool,
  required: PropTypes.bool,
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  info: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  help: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  margin: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  pad: PropTypes.bool,
};

Select.defaultProps = {
  dropHeight: 'small',
  replace: true,
};

export default Select;
