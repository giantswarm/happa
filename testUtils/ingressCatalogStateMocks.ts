import { Constants } from 'shared';
import { appCatalogsResponse } from 'testUtils/mockHttpCalls';

const ingressCatalog = appCatalogsResponse.find(
  (c) => c.metadata.name === Constants.INSTALL_INGRESS_TAB_APP_CATALOG_NAME
);
export const catalogsState = {
  entities: {
    catalogs: {
      items: {
        [Constants.INSTALL_INGRESS_TAB_APP_CATALOG_NAME]: ingressCatalog,
      },
    },
  },
};
