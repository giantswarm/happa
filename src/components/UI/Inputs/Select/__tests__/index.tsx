import { renderWithTheme } from 'testUtils/renderUtils';

import Select from '..';

describe('Select', () => {
  it('renders a simple input', () => {
    const { container } = renderWithTheme(Select, {
      id: 'pet',
      options: ['A dog', 'A cat', 'A frog'],
      placeholder: 'Just select something...',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders an input with a label', () => {
    const { container } = renderWithTheme(Select, {
      id: 'pet',
      options: ['A dog', 'A cat', 'A frog'],
      placeholder: 'Just select something...',
      label: 'Which pet to get?',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders an input with a validation error', () => {
    const { container } = renderWithTheme(Select, {
      id: 'pet',
      options: ['A dog', 'A cat', 'A frog'],
      placeholder: 'Just select something...',
      label: 'Which pet to get?',
      error: 'You failed at this',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders an input with a help message', () => {
    const { container } = renderWithTheme(Select, {
      id: 'pet',
      options: ['A dog', 'A cat', 'A frog'],
      placeholder: 'Just select something...',
      label: 'Which pet to get?',
      help: 'This is a very cool input',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders an input with an info message', () => {
    const { container } = renderWithTheme(Select, {
      id: 'pet',
      options: ['A dog', 'A cat', 'A frog'],
      placeholder: 'Just select something...',
      label: 'Which pet to get?',
      info: 'You can set a cool value',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders a disabled input', () => {
    const { container } = renderWithTheme(Select, {
      id: 'pet',
      options: ['A dog', 'A cat', 'A frog'],
      placeholder: 'Just select something...',
      value: 'A dog',
      label: 'Which pet to get?',
      disabled: true,
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
