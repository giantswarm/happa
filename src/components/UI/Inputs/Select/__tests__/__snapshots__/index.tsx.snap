// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders a disabled input 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 yBlmn"
>
  <div
    class="ThemeProvider__AppWrapper-sc-1qc9t9l-0 breHlv"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jazqMc FormField__FormFieldBox-m9hood-0 jGEMXj"
    >
      <label
        class="StyledText-sc-1sadyjn-0 esuSAl"
        for="pet"
      >
        Which pet to get?
      </label>
      <div
        class="StyledBox-sc-13pk1d4-0 fwONFA FormField__FormFieldContentBox-m9hood-1"
      >
        <button
          aria-label="Open Drop"
          class="StyledButton-sc-323bzc-0 gqtTVF Select__StyledSelectDropButton-sc-17idtfo-1 hNACHY"
          disabled=""
          id="pet"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bRsrZW"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 iAryDB"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 jaFOFt Select__SelectTextInput-sc-17idtfo-0 cANsmG"
                  id="pet__input"
                  placeholder="Just select something..."
                  readonly=""
                  tabindex="-1"
                  type="text"
                  value="A dog"
                />
              </div>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 kmHwPP"
              style="min-width: auto;"
            >
              <svg
                aria-label="FormDown"
                class="StyledIcon-ofa7kd-0 deqPrt"
                viewBox="0 0 24 24"
              >
                <polyline
                  fill="none"
                  points="18 9 12 15 6 9"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders a simple input 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 yBlmn"
>
  <div
    class="ThemeProvider__AppWrapper-sc-1qc9t9l-0 breHlv"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jazqMc FormField__FormFieldBox-m9hood-0 jGEMXj"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 drdhXT FormField__FormFieldContentBox-m9hood-1"
      >
        <button
          aria-label="Open Drop"
          class="StyledButton-sc-323bzc-0 hkpJCj Select__StyledSelectDropButton-sc-17idtfo-1 hNACHY"
          id="pet"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bRsrZW"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 iAryDB"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 jaFOFt Select__SelectTextInput-sc-17idtfo-0 bIurki"
                  id="pet__input"
                  placeholder="Just select something..."
                  readonly=""
                  tabindex="-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 kmHwPP"
              style="min-width: auto;"
            >
              <svg
                aria-label="FormDown"
                class="StyledIcon-ofa7kd-0 deqPrt"
                viewBox="0 0 24 24"
              >
                <polyline
                  fill="none"
                  points="18 9 12 15 6 9"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders an input inside the dropdown, that can search through the options 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 yBlmn"
>
  <div
    class="ThemeProvider__AppWrapper-sc-1qc9t9l-0 breHlv"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jazqMc FormField__FormFieldBox-m9hood-0 jGEMXj"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fwONFA FormField__FormFieldContentBox-m9hood-1"
      >
        <button
          aria-label="Open Drop"
          class="StyledButton-sc-323bzc-0 gqtTVF Select__StyledSelectDropButton-sc-17idtfo-1 hNACHY"
          disabled=""
          id="pet"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bRsrZW"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 iAryDB"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 jaFOFt Select__SelectTextInput-sc-17idtfo-0 cANsmG"
                  id="pet__input"
                  readonly=""
                  tabindex="-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 kmHwPP"
              style="min-width: auto;"
            >
              <svg
                aria-label="FormDown"
                class="StyledIcon-ofa7kd-0 deqPrt"
                viewBox="0 0 24 24"
              >
                <polyline
                  fill="none"
                  points="18 9 12 15 6 9"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders an input with a help message 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 yBlmn"
>
  <div
    class="ThemeProvider__AppWrapper-sc-1qc9t9l-0 breHlv"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jazqMc FormField__FormFieldBox-m9hood-0 jGEMXj"
    >
      <label
        class="StyledText-sc-1sadyjn-0 esuSAl"
        for="pet"
      >
        Which pet to get?
      </label>
      <span
        class="StyledText-sc-1sadyjn-0 erMGdB"
      >
        This is a very cool input
      </span>
      <div
        class="StyledBox-sc-13pk1d4-0 drdhXT FormField__FormFieldContentBox-m9hood-1"
      >
        <button
          aria-label="Open Drop"
          class="StyledButton-sc-323bzc-0 hkpJCj Select__StyledSelectDropButton-sc-17idtfo-1 hNACHY"
          id="pet"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bRsrZW"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 iAryDB"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 jaFOFt Select__SelectTextInput-sc-17idtfo-0 bIurki"
                  id="pet__input"
                  placeholder="Just select something..."
                  readonly=""
                  tabindex="-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 kmHwPP"
              style="min-width: auto;"
            >
              <svg
                aria-label="FormDown"
                class="StyledIcon-ofa7kd-0 deqPrt"
                viewBox="0 0 24 24"
              >
                <polyline
                  fill="none"
                  points="18 9 12 15 6 9"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders an input with a label 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 yBlmn"
>
  <div
    class="ThemeProvider__AppWrapper-sc-1qc9t9l-0 breHlv"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jazqMc FormField__FormFieldBox-m9hood-0 jGEMXj"
    >
      <label
        class="StyledText-sc-1sadyjn-0 esuSAl"
        for="pet"
      >
        Which pet to get?
      </label>
      <div
        class="StyledBox-sc-13pk1d4-0 drdhXT FormField__FormFieldContentBox-m9hood-1"
      >
        <button
          aria-label="Open Drop"
          class="StyledButton-sc-323bzc-0 hkpJCj Select__StyledSelectDropButton-sc-17idtfo-1 hNACHY"
          id="pet"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bRsrZW"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 iAryDB"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 jaFOFt Select__SelectTextInput-sc-17idtfo-0 bIurki"
                  id="pet__input"
                  placeholder="Just select something..."
                  readonly=""
                  tabindex="-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 kmHwPP"
              style="min-width: auto;"
            >
              <svg
                aria-label="FormDown"
                class="StyledIcon-ofa7kd-0 deqPrt"
                viewBox="0 0 24 24"
              >
                <polyline
                  fill="none"
                  points="18 9 12 15 6 9"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders an input with a validation error 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 yBlmn"
>
  <div
    class="ThemeProvider__AppWrapper-sc-1qc9t9l-0 breHlv"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jazqMc FormField__FormFieldBox-m9hood-0 jGEMXj"
    >
      <label
        class="StyledText-sc-1sadyjn-0 esuSAl"
        for="pet"
      >
        Which pet to get?
      </label>
      <div
        class="StyledBox-sc-13pk1d4-0 dfycIj FormField__FormFieldContentBox-m9hood-1"
      >
        <button
          aria-label="Open Drop"
          class="StyledButton-sc-323bzc-0 hkpJCj Select__StyledSelectDropButton-sc-17idtfo-1 hNACHY"
          id="pet"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bRsrZW"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 iAryDB"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 jaFOFt Select__SelectTextInput-sc-17idtfo-0 bIurki"
                  id="pet__input"
                  placeholder="Just select something..."
                  readonly=""
                  tabindex="-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 kmHwPP"
              style="min-width: auto;"
            >
              <svg
                aria-label="FormDown"
                class="StyledIcon-ofa7kd-0 deqPrt"
                viewBox="0 0 24 24"
              >
                <polyline
                  fill="none"
                  points="18 9 12 15 6 9"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
      <span
        class="StyledText-sc-1sadyjn-0 dxYrxD"
      >
        You failed at this
      </span>
    </div>
  </div>
</div>
`;

exports[`Select renders an input with an info message 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 yBlmn"
>
  <div
    class="ThemeProvider__AppWrapper-sc-1qc9t9l-0 breHlv"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jazqMc FormField__FormFieldBox-m9hood-0 jGEMXj"
    >
      <label
        class="StyledText-sc-1sadyjn-0 esuSAl"
        for="pet"
      >
        Which pet to get?
      </label>
      <div
        class="StyledBox-sc-13pk1d4-0 drdhXT FormField__FormFieldContentBox-m9hood-1"
      >
        <button
          aria-label="Open Drop"
          class="StyledButton-sc-323bzc-0 hkpJCj Select__StyledSelectDropButton-sc-17idtfo-1 hNACHY"
          id="pet"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bRsrZW"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 iAryDB"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 jaFOFt Select__SelectTextInput-sc-17idtfo-0 bIurki"
                  id="pet__input"
                  placeholder="Just select something..."
                  readonly=""
                  tabindex="-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 kmHwPP"
              style="min-width: auto;"
            >
              <svg
                aria-label="FormDown"
                class="StyledIcon-ofa7kd-0 deqPrt"
                viewBox="0 0 24 24"
              >
                <polyline
                  fill="none"
                  points="18 9 12 15 6 9"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
      <span
        class="StyledText-sc-1sadyjn-0 fmFxNm"
      >
        You can set a cool value
      </span>
    </div>
  </div>
</div>
`;
